suitcase/pyxrf/export.py

```diff
@@ -0,0 +1,349 @@
+        for i in range(pos_data.shape[2]):
+            # flip position the same as data flip on det counts
+            pos_data[:, :, i] = flip_data(pos_data_temp[:, :, i], subscan_dims=subscan_dims)
+    for i, v in enumerate(pos_names):
+        data_output[v] = pos_data[:, :, i]
+
+    # scaler data
+    scaler_names, scaler_data = get_name_value_from_db(scaler_list, data,
+                                                       datashape)
+    if fly_type in ('pyramid',):
+        scaler_data = flip_data(scaler_data, subscan_dims=subscan_dims)
+    for i, v in enumerate(scaler_names):
+        data_output[v] = scaler_data[:, :, i]
+    return data_output
+
+
+def write_db_to_hdf_base(fpath, data, num_det=3, create_each_det=True):
+    """
+    Data is a dictionary of numpy array. Save the data to hdf file.
+
+    Parameters
+    ----------
+    fpath: str
+        path to save hdf file
+    data : dict
+        fluorescence data with scaler value and positions
+    num_det : int
+        number of detector
+    create_each_det : Bool, optional
+        if number of point is too large, only sum data is saved in h5 file
+    """
+    interpath = 'xrfmap'
+    sum_data = None
+
+    with h5py.File(fpath, 'a') as f:
+        if create_each_det is True:
+            for n in range(num_det):
+                detname = 'det' + str(n+1)
+                new_data = data[detname]
+
+                if sum_data is None:
+                    sum_data = new_data
+                else:
+                    sum_data += new_data
+
+                dataGrp = f.create_group(interpath+'/'+detname)
+                ds_data = dataGrp.create_dataset('counts', data=new_data, compression='gzip')
+                ds_data.attrs['comments'] = 'Experimental data from channel ' + str(n)
+        else:
+            sum_data = data['det_sum']
+
+        # summed data
+        if sum_data is not None:
+            dataGrp = f.create_group(interpath+'/detsum')
+            ds_data = dataGrp.create_dataset('counts', data=sum_data, compression='gzip')
+            ds_data.attrs['comments'] = 'Experimental data from channel sum'
+
+        # add positions
+        if 'pos_names' in data:
+            dataGrp = f.create_group(interpath+'/positions')
+            pos_names = data['pos_names']
+            pos_data = data['pos_data']
+            dataGrp.create_dataset('name', data=helper_encode_list(pos_names))
+            dataGrp.create_dataset('pos', data=pos_data)
+
+        # scaler data
+        if 'scaler_data' in data:
+            dataGrp = f.create_group(interpath+'/scalers')
+            scaler_names = data['scaler_names']
+            scaler_data = data['scaler_data']
+            dataGrp.create_dataset('name', data=helper_encode_list(scaler_names))
+            dataGrp.create_dataset('val', data=scaler_data)
+
+
+def get_name_value_from_db(name_list, data, datashape):
+    """
+    Get data from db, and parse it to the required format and output.
```

Comment:
I would rephrase the docstring and maybe rename the function. "Get data from the db" sounds like it fetch data from the DataBroker.

Reply:
I will add more doc. thanks for suggestion.